lib/internal/util/comparisons.js

```diff
@@ -767,11 +788,28 @@ function sparseArrayEquiv(a, b, mode, memos, i) {
   return true;
 }
 
-function objEquiv(a, b, mode, keys2, memos, iterationType) {
+function objEquiv(a, b, mode, keys1, keys2, memos, iterationType) {
   // The pair must have equivalent values for every corresponding key.
   if (keys2.length > 0) {
-    for (const key of keys2) {
-      if (!innerDeepEqual(a[key], b[key], mode, memos)) {
+    let i = 0;
+    // Ordered keys
+    if (keys1 !== undefined) {
+      for (; i < keys2.length; i++) {
+        const key = keys2[i];
+        if (keys1[i] !== key) {
+          break;
+        }
+        if (!innerDeepEqual(a[key], b[key], mode, memos)) {
+          return false;
+        }
+      }
+    }
+    // Unordered keys
+    for (; i < keys2.length; i++) {
+      const key = keys2[i];
+      const descriptor = ObjectGetOwnPropertyDescriptor(a, key);
+      if (!descriptor?.enumerable ||
+          !innerDeepEqual(descriptor.value !== undefined ? descriptor.value : a[key], b[key], mode, memos)) {
```

Comment:
if you're only checking enumerability + the value, is it faster to get the descriptor, or to use `ObjectPrototypePropertyIsEnumerable` and `a[key]`?

Reply:
This one is an interesting one. I would have expected that `isEnumerable` + `a[key]` is faster, but I thought I might just check what happens when I get the whole descriptor. It turned out that this code is faster by about 3%.